cpp/src/neighbors/all_neighbors/all_neighbors_merge.cuh

```diff
@@ -0,0 +1,266 @@
+/*
```

Comment:
this file is mostly a copy of functions in the existing `cuvs/cpp/src/neighbors/detail/nn_descent_batch.cuh` file ([link](https://github.com/rapidsai/cuvs/blob/branch-25.06/cpp/src/neighbors/detail/nn_descent_batch.cuh))